numpy/lib/npyio.py

```diff
@@ -1104,7 +1104,7 @@ def tobytes_first(x, conv):
                 nshape = list(X.shape)
                 pos = nshape[0]
                 nshape[0] += len(x)
-                X.resize(nshape)
+                X.resize(nshape, refcheck=False)
                 X[pos:, ...] = x
```

Comment:
This code looks very dubious to me -- the comment claims that the overhead from `resize` should be small compared to the cost of reading the data, but reading the data is O(n) and resizing is O(n**2), so for large enough arrays it will always dominate, right? I guess for large arrays we'll usually get lucky with the memory allocator, but still...

Anyway, that's not this PR's fault, and the actual change looks correct.

Reply:
Right, it should probably just do the standard exponential overallocation here.